unity/Assets/Scripts/AgentManager.cs

```diff
@@ -1846,14 +1868,21 @@ public class MultiAgentMetadata {
 [MessagePackObject(keyAsPropertyName: true)]
 public class ThirdPartyCameraMetadata {
     public int thirdPartyCameraId;
-    public Vector3 position;
-    public Vector3 rotation;
+    public Vector3 position; //to be deprecated
+    public Vector3 rotation; //to be deprecated
+    public Vector3 worldRelativeThirdPartyCameraPosition;
+    public Vector3 worldRelativeThirdPartyCameraRotation;
     public float fieldOfView;
 
     //note these should only be returned with values
     //if the third party camera is a child of the agent
-    public Vector3 agentPositionRelativeThirdPartyCameraPosition;
-    public Vector3 agentPositionRelativeThirdPartyCameraRotation;
+    public Vector3? agentPositionRelativeThirdPartyCameraPosition;
+    public Vector3? agentPositionRelativeThirdPartyCameraRotation;
+
+    //return the local space coordinates if this third party camera has a parent object, this may be the same as agentPositionRelative depending on how things are parented
+    public Vector3? parentPositionRelativeThirdPartyCameraPosition;
+    public Vector3? parentPositionRelativeThirdPartyCameraRotation;
+    public string parentObjectName; //if this third party camera is in a heirarchy, return the name of the parent object
```

Comment:
"Hierarchy" is misspelled

Reply:
fixed in ef3bfb20c08ec9a7a991b23d7e9008af3c72d132